NullReferenceException at Mono.Linker.Steps.MarkStep.SearchPropertiesForMatchingFieldDefinition.

Dear, after update Visual Studio 2017 to 15.5.2 when I try generate a new Android release of my app I get this error:

error MSB4018: The "LinkAssemblies" task failed unexpectedly.
error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object.
error MSB4018: at Mono.Linker.Steps.MarkStep.SearchPropertiesForMatchingFieldDefinition(FieldDefinition field)
error MSB4018: at Mono.Linker.Steps.MarkStep.MarkFields(TypeDefinition type, Boolean includeStatic, Boolean markBackingFieldsOnlyIfPropertyMarked)
error MSB4018: at Mono.Linker.Steps.MarkStep.ApplyPreserveInfo(TypeDefinition type)
error MSB4018: at Mono.Linker.Steps.MarkStep.MarkType(TypeReference reference)
error MSB4018: at MonoDroid.Tuner.MonoDroidMarkStep.MarkType(TypeReference reference)
error MSB4018: at Mono.Linker.Steps.MarkStep.InitializeType(TypeDefinition type)
error MSB4018: at Mono.Linker.Steps.MarkStep.InitializeAssembly(AssemblyDefinition assembly)
error MSB4018: at Mono.Linker.Steps.MarkStep.Initialize()
error MSB4018: at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
error MSB4018: at Mono.Linker.Pipeline.Process(LinkContext context)
error MSB4018: at MonoDroid.Tuner.Linker.Process(LinkerOptions options, LinkContext& context)
error MSB4018: at Xamarin.Android.Tasks.LinkAssemblies.Execute(DirectoryAssemblyResolver res)
error MSB4018: at Xamarin.Android.Tasks.LinkAssemblies.Execute()
error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.d__26.MoveNext()

I already reported the issue on VS community (https://developercommunity.visualstudio.com/content/problem/175090/the-linkassemblies-task-failed-unexpectedly-with-v.html), but, anybody know some productive way of I investigate/debug this issue? How can I collect more clues about the problem? Although configured VS in diagnostic mode, I do not see any useful information in the logs that might indicate the motivation for the problem.

Are there some way of detect what the specific motivation of the exception?

--- a/linker/Linker.Steps/MarkStep.cs
+++ b/linker/Linker.Steps/MarkStep.cs
@@ -1227,7 +1227,7 @@ namespace Mono.Linker.Steps {
 				if (!includeStatic && field.IsStatic)
 					continue;
 
-				if (markBackingFieldsOnlyIfPropertyMarked && field.Name.EndsWith (">k__BackingField")) {
+				if (markBackingFieldsOnlyIfPropertyMarked && field.Name.EndsWith (">k__BackingField", StringComparison.Ordinal)) {
 					// We can't reliably construct the expected property name from the backing field name for all compilers
 					// because csc shortens the name of the backing field in some cases
 					// For example:
@@ -1246,8 +1246,12 @@ namespace Mono.Linker.Steps {
 		static PropertyDefinition SearchPropertiesForMatchingFieldDefinition (FieldDefinition field)
 		{
 			foreach (var property in field.DeclaringType.Properties) {
-				foreach (var ins in property.GetMethod.Body.Instructions) {
-					if (ins.Operand != null && ins.Operand == field)
+				var instr = property.GetMethod?.Body?.Instructions;
+				if (instr == null)
+					continue;
+
+				foreach (var ins in instr) {
+					if (ins?.Operand == field)
 						return property;
 				}
 			}

--- a/linker/Tests/Mono.Linker.Tests.Cases/LinkXml/PreserveBackingFieldWhenPropertyIsKept.cs
+++ b/linker/Tests/Mono.Linker.Tests.Cases/LinkXml/PreserveBackingFieldWhenPropertyIsKept.cs
@@ -1,0 +1,18 @@
+﻿using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.LinkXml
+{
+	abstract class PreserveBackingFieldWhenPropertyIsKept
+	{
+		public static void Main ()
+		{
+			Prop = 1;
+		}
+
+		public abstract int Base { set; }
+
+		[Kept]
+		[KeptBackingField]
+		public static int Prop { get; [Kept] set; }		
+	}
+}

--- a/linker/Tests/Mono.Linker.Tests.Cases/LinkXml/PreserveBackingFieldWhenPropertyIsKept.xml
+++ b/linker/Tests/Mono.Linker.Tests.Cases/LinkXml/PreserveBackingFieldWhenPropertyIsKept.xml
@@ -1,0 +1,6 @@
+﻿<linker>
+  <assembly fullname="test, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
+    <type fullname="Mono.Linker.Tests.Cases.LinkXml.PreserveBackingFieldWhenPropertyIsKept" preserve="fields">
+    </type>
+  </assembly>
+</linker>

--- a/linker/Tests/Mono.Linker.Tests.Cases/Mono.Linker.Tests.Cases.csproj
+++ b/linker/Tests/Mono.Linker.Tests.Cases/Mono.Linker.Tests.Cases.csproj
@@ -165,6 +165,7 @@
     <Compile Include="TypeForwarding\MissingTargetReference.cs" />
     <None Include="TypeForwarding\Dependencies\TypeForwarderMissingReference.il" />
     <Compile Include="Basic\UsedInterfaceIsKept.cs" />
+    <Compile Include="LinkXml\PreserveBackingFieldWhenPropertyIsKept.cs" />
   </ItemGroup>
   <ItemGroup>
     <Content Include="LinkXml\CanPreserveTypesUsingRegex.xml" />
@@ -192,6 +193,7 @@
     <Content Include="Resources\Dependencies\EmbeddedLinkXmlFileIsProcessed.xml" />
     <Content Include="Resources\Dependencies\NonLinkerEmbeddedResourceHasNoImpact.xml" />
     <Content Include="TestFramework\Dependencies\VerifyResourceInAssemblyAttributesBehavior.txt" />
+    <Content Include="LinkXml\PreserveBackingFieldWhenPropertyIsKept.xml" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\Mono.Linker.Tests.Cases.Expectations\Mono.Linker.Tests.Cases.Expectations.csproj">

